spring-cloud-gcp-core/src/main/java/org/springframework/cloud/gcp/core/util/MapBuilder.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.core.util;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
```

Comment:
Can you add a javadoc here to clarify purpose of class and intention to remove with Java 9?

Reply:
Yeah, I just wanted to make sure this was okay to add to `spring-cloud-gcp-core` first.
@artembilan @dmitry-s @ChengyuanZhao What do you think?

Reply:
Looks good.